crud.c

```diff
@@ -0,0 +1,179 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+// structure of user
+typedef struct {
+    int id;
+    char name[50];
+    int age;
+} User;
+
+// creates file to store user data
+void createFile() {
+    FILE *file = fopen("users.txt", "w");
+    if (file == NULL) {
+        printf("Error in creating the file.\n");
+        return;
+    }
+    fclose(file);
+}
+
+// Function to add a user
+void addUser() {
+    User user;
+    FILE *file = fopen("users.txt", "a");
+    if (!file) {
+        printf("Error in opening the file.\n");
+        return;
+    }
+
+    printf("Enter user ID: ");
+    scanf("%d", &user.id);
+    printf("Enter user name: ");
+    getchar();
+    fgets(user.name, sizeof(user.name), stdin);
+    user.name[strcspn(user.name, "\n")] = '\0';
+    printf("Enter user age: ");
+    scanf("%d", &user.age);
+
+    fprintf(file, "%d,%s,%d\n", user.id, user.name, user.age);
+    fclose(file);
+    printf("User added successfully.\n");
+}
+
+// Function to display all users
+void displayUsers() {
+    FILE *file = fopen("users.txt", "r");
+    if (!file) {
+        printf("Error opening the file.\n");
+        return;
+    }
+
+    User user;
+    printf("ID\tName\tAge\n");
+    printf("-----------------------\n");
+
+    while (fscanf(file, "%d,%49[^,],%d\n", &user.id, user.name, &user.age) != EOF) {
+        printf("%d\t%s\t%d\n", user.id, user.name, user.age);
+    }
+    fclose(file);
+}
+
+// Function to update a user
+void updateUser(int id) {
+    User user;
+    int found = 0;
+    FILE *file = fopen("users.txt", "r");
+    if (!file) {
+        printf("Error opening file.\n");
+        return;
+    }
+
+    FILE *tempFile = fopen("temp.txt", "w");
+    if (!tempFile) {
+        printf("Error creating temporary file.\n");
+        fclose(file);
+        return;
+    }
+
+    while (fscanf(file, "%d,%49[^,],%d\n", &user.id, user.name, &user.age) != EOF) {
+        if (user.id == id) {
+            found = 1;
+            printf("Enter new user ID: ");
+            scanf("%d", &user.id);
+            printf("Enter new user name: ");
+            getchar();
+            fgets(user.name, sizeof(user.name), stdin);
+            user.name[strcspn(user.name, "\n")] = '\0';
+            printf("Enter new user age: ");
+            scanf("%d", &user.age);
+        }
+        fprintf(tempFile, "%d,%s,%d\n", user.id, user.name, user.age);
+    }
+
+    fclose(file);
+    fclose(tempFile);
+
+    if (found) {
+        remove("users.txt");
```

Comment:
Use only one txt file. It's not necessary to maintain the user details after the program termination.

Reply:
okay!